pymc/distributions/continuous.py

```diff
@@ -835,6 +838,9 @@ def logcdf(value, loc, sigma):
             msg="sigma > 0",
         )
 
+    def icdf(value, lower, upper, loc, sigma):
+        return stats.truncnorm.ppf(q=value, a=lower, b=upper, loc=loc, scale=sigma)
```

Comment:
The `icdf` has to have the same signature as the `logp` and `logcdf` of the same distribution. 

Also you can't call `scipy` or `numpy` routines, the expression has to be implemented exclusively in terms of pytensor operations

Reply:
a trial icdf for Wald dist already commited